rrs/tools/rrs_upstream_history.py

```diff
@@ -0,0 +1,240 @@
+    """
+        Utility function to set regexes to SPECIAL_PKGSUFFIX packages
+        that don't have set it.
+
+        For example: python-native use regex from python if don't have
+        one set it.
+    """
+
+    variables = ('UPSTREAM_CHECK_REGEX', 'UPSTREAM_CHECK_URI', 'UPSTREAM_CHECK_GITTAGREGEX')
+
+    if any(d.getVar(var, True) for var in variables):
+        return
+
+    suffixes = d.getVar('SPECIAL_PKGSUFFIX', True).split()
+    prefixes = ['nativesdk-']
+
+    special = list(suffixes)
+    special.extend(prefixes)
+
+    localdata = bb.data.createCopy(d)
+    pn = localdata.getVar('PN', True)
+    for s in special:
+        if pn.find(s) != -1:
+            if s in suffixes:
+                pnstripped = pn.split(s)[0]
+            else:
+                pnstripped = pn.replace(s, '')
+
+            localdata.setVar('OVERRIDES', "pn-" + pnstripped + ":" +
+                    d.getVar('OVERRIDES', True))
+            bb.data.update_data(localdata)
+
+            for var in variables:
+                new_value = localdata.getVar(var, True)
+                if new_value is None:
+                    continue
+
+                d.setVar(var, new_value)
+                logger.debug("%s: %s new value %s" % (pn, var,
+                    d.getVar(var, True)))
+            break
+
+def get_upstream_info(layerbranch, recipe_data, result):
+    from bb.utils import vercmp_string
+    from oe.recipeutils import get_recipe_upstream_version, \
+            get_recipe_pv_without_srcpv
+
+    pn = recipe_data.getVar('PN', True)
+
+    ru = RecipeUpstream()
+    summary = recipe_data.getVar('SUMMARY', True) or recipe_data.getVar('DESCRIPTION', True)
+    ru.recipesymbol = RecipeSymbol.symbol(pn, layerbranch, summary=summary)
+    recipe_pv = recipe_data.getVar('PV', True)
+
+    ru_info = None
+    try:
+        ru_info = get_recipe_upstream_version(recipe_data)
+    except Exception as e:
+        logger.exception("%s: in layer branch %s, %s" % (pn,
+            str(layerbranch), str(e)))
+
+    if ru_info is not None and ru_info['version']:
+        ru.version = ru_info['version']
+        ru.type = ru_info['type']
+        ru.date = ru_info['datetime']
+
+        pv, _, _ = get_recipe_pv_without_srcpv(recipe_pv,
+                get_pv_type(recipe_pv)) 
+        upv, _, _ = get_recipe_pv_without_srcpv(ru_info['version'],
+                get_pv_type(ru_info['version']))
+
+        if pv and upv:
+            cmp_ver = vercmp_string(pv, upv)
+            if cmp_ver == -1:
+                ru.status = 'N' # Not update
+            elif cmp_ver == 0:
+                ru.status = 'Y' # Up-to-date
+            elif cmp_ver == 1:
+                ru.status = 'D' # Downgrade, need to review why
+        else:
+            logger.debug('Unable to determine upgrade status for %s: %s -> %s' % (pn, pv, upv))
+            ru.status = 'U' # Unknown
+    else:
+        ru.version = ''
+        ru.type = 'M'
+        ru.date = datetime.now()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The naive datetime objects are treated by many datetime methods as local times, it is preferred to use aware datetimes to represent times in UTC. The recommended way to create an aware datetime object representing a specific timestamp in UTC is by passing `tzinfo` as an argument to the method.

[Learn more](https://docs.python.org/3/library/datetime.html#aware-and-naive-objects)